### PR TITLE
Fix/api student form

### PR DESCRIPTION
**Pull request by stompGT**
## Proposed Changes

- Bring back Students::CreateForm to be used only be API. This form was removed by c8a36a658391e833723c497430c99ec231b2eea7
- Fix some of Keycloak::FakeClient methods

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Check if route, query, or mutation authorization looks correct.
  - Add tests for authorization, if required.
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Update developer and product docs, where applicable.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Check if new tables or columns that have been added need to be handled in the following services:
  - `Users::DeleteAccountService`
  - `Courses::CloneService`
  - `Courses::DeleteService`
  - `Courses::DemoContentService`
  - `Schools::DeleteService`
- [ ] Check if changes in _packaged_ components have been published to `npm`.
- [ ] Add development seeds for new tables.

### CODE REVIEW

File: app/forms/api/schools/students/create_form.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def save
         private
 
         def notify?
-          notify == 'true'
+          notify.to_s == 'true'
```
Comment (mpraglowski) [inline app/forms/api/schools/students/create_form.rb:32]: Ekhm... what? why? could you explain why we do this like that here?

File: lib/keycloak.rb [hunk 2]
```diff
@@ -255,8 +255,7 @@ def user_signed_in?(access_token)
 
     def user_token(uname_email, password)
       user = user_by_email(uname_email)
-      valid = user && user.dig(:credentials, :password) == password
-      raise FailedRequestError.new 'Failed to set user password' unless valid
+      valid = user && user.dig(:credentials, :value) == password
```
Comment (mpraglowski) [inline lib/keycloak.rb:258]: Why bother checking if we do nothing with the result? I do not understand that change.

File: lib/keycloak.rb [hunk 3]
```diff
@@ -269,7 +268,7 @@ def user_sign_out(refresh_token)
 
     private
     def user_by_token(token)
-      @users.find{|_, user| user[:token] == token}
+      @users.find{|_, user| user[:token] == token}[1]
```
Comment (mpraglowski) [inline lib/keycloak.rb:272]:
```ruby
      @users[email] = {
        username: email,
        email: email,
        firstName: first_name,
        lastName: last_name,
        enabled: true,
        active: true,
        credentials: {},
      }.with_indifferent_access
```

what will do `[1]` on results here ?

File: lib/keycloak.rb (hunk 2 above)
Comment (mpraglowski) [reply lib/keycloak.rb:258]: ok - the check is back in 1c2025a7e4da887845d511ae8019d29e2316a3b9 ;)

**Comment by stompGT** [comment]
> "Bring back Student create form to be used under API scope" ? Bring back from what? I see no earlier history of this form.

This [PR](https://github.com/pupilfirst/pupilfirst/pull/579/files) from PF team removed `Schools::Founders::CreateForm` and moved this logic into `CreateStudentsMutator`. The API relied on the `CreateForm` to perform the validation of the params. When they merged the PR and that was merged into out master it broke the integration. As `CreateStudentMutator` is rely on GraphQL to build the context and future changes to that Mutator could break our integration, i though it would be better to copy the old form and bring it to the context of the API.

**Comment by mpraglowski** [comment]
included in #57

**Pull request closed without merging**